packages/backend/src/storage/storage.ts

```diff
@@ -506,29 +514,56 @@ export class Storage {
       })
     }
 
-    writeStream.end()
+    if (downloadState === DownloadState.Canceled) {
+      const downloadCanceled: DownloadProgress = {
+        size: metadata.size,
+        downloaded: 0,
+        transferSpeed: 0
+      }
 
-    const downloadCompleted: DownloadProgress = {
-      size: metadata.size,
-      downloaded: metadata.size,
-      transferSpeed: 0
-    }
+      const statusCanceled: DownloadStatus = {
+        cid: metadata.cid,
+        downloadState: DownloadState.Canceled,
+        downloadProgress: downloadCanceled
+      }
 
-    const statusCompleted: DownloadStatus = {
-      cid: metadata.cid,
-      downloadState: DownloadState.Completed,
-      downloadProgress: downloadCompleted
-    }
+      // Canceled Download
+      this.io.updateDownloadProgress(statusCanceled)
 
-    // Downloaded file
-    this.io.updateDownloadProgress(statusCompleted)
+      // Clear cancellation signal
+      const index = this.downloadCancellations.indexOf(metadata.cid)
+      if (index > -1) {
```

Comment:
is it not better `index !== -1 ` ?

Reply:
`index > -1` is less tolerant also `index !== -1` is true for index equal undefined

Reply:
or replace 
`const index = this.downloadCancellations.indexOf(metadata.cid)
      if (index > -1) {`
with
`if(this.downloadCancellations.includes(metadata.cid)){`

Reply:
You need to know the index of an element in array to be able to remove it. So yeah, you can do it by using `includes` method but it'll only give you additional line of code.
Also it's always safe to assume that some value can be `undefined` or `null`, especially when it doesn't force you to change the underlaying logic